// @create-figma-plugin/ui/lib/components/dropdown-menu/dropdown-menu.stories.js
/** @jsx h */
import { h } from 'preact';
import { useState } from 'preact/hooks';
import { DropdownMenu } from './dropdown-menu';
export default { title: 'Dropdown Menu' };
const options = [
    { value: 'foo' },
    { value: 'bar' },
    { value: 'baz' },
    { separator: true },
    { header: 'Header' },
    { value: 'qux' },
    { value: 'quux' },
    { value: 'quuux' },
    { value: 'quuuux' },
    { value: 'quuuuux' }
];
function Foo({ foo }) {
    return h("h1", { style: { width: '240px' } }, foo === null ? 'Select…' : foo);
}
export const BottomLeft = function () {
    const [state, setState] = useState({ foo: null });
    return (h(DropdownMenu, { name: "foo", onChange: setState, options: options, value: state.foo },
        h(Foo, null)));
};
export const BottomRight = function () {
    const [state, setState] = useState({ foo: null });
    return (h(DropdownMenu, { name: "foo", onChange: setState, options: options, right: true, value: state.foo },
        h(Foo, null)));
};
export const TopLeft = function () {
    const [state, setState] = useState({ foo: null });
    return (h("div", { style: {
            marginTop: '128px'
        } },
        h(DropdownMenu, { name: "foo", onChange: setState, options: options, top: true, value: state.foo },
            h(Foo, null))));
};
export const TopRight = function () {
    const [state, setState] = useState({ foo: null });
    return (h("div", { style: {
            marginTop: '128px'
        } },
        h(DropdownMenu, { name: "foo", onChange: setState, options: options, right: true, top: true, value: state.foo },
            h(Foo, null))));
};
export const Filled = function () {
    const [state, setState] = useState({ foo: 'qux' });
    return (h(DropdownMenu, { name: "foo", onChange: setState, options: options, value: state.foo },
        h(Foo, null)));
};
export const FullWidth = function () {
    const [state, setState] = useState({ foo: null });
    return (h(DropdownMenu, { fullWidth: true, name: "foo", onChange: setState, options: options, value: state.foo },
        h(Foo, null)));
};
