// @create-figma-plugin/ui/lib/components/button/button.stories.js
/** @jsx h */
import { h } from 'preact';
import { Button } from './button';
export default { title: 'Button' };
const longText = Array(100).fill('Text').join(' ');
function handleClick() { }
export const Primary = function () {
    return h(Button, { onClick: handleClick }, "Text");
};
export const PrimaryLoading = function () {
    return (h(Button, { loading: true, onClick: handleClick }, "Text"));
};
export const PrimaryDisabled = function () {
    return (h(Button, { disabled: true, onClick: handleClick }, "Text"));
};
export const PrimaryDisabledLoading = function () {
    return (h(Button, { disabled: true, loading: true, onClick: handleClick }, "Text"));
};
export const PrimaryFullWidth = function () {
    return (h(Button, { fullWidth: true, onClick: handleClick }, "Text"));
};
export const PrimaryFullWidthLongText = function () {
    return (h(Button, { fullWidth: true, onClick: handleClick }, longText));
};
export const Secondary = function () {
    return (h(Button, { onClick: handleClick, secondary: true }, "Text"));
};
export const SecondaryLoading = function () {
    return (h(Button, { loading: true, onClick: handleClick, secondary: true }, "Text"));
};
export const SecondaryDisabled = function () {
    return (h(Button, { disabled: true, onClick: handleClick, secondary: true }, "Text"));
};
export const SecondaryDisabledLoading = function () {
    return (h(Button, { disabled: true, loading: true, onClick: handleClick, secondary: true }, "Text"));
};
export const SecondaryFullWidth = function () {
    return (h(Button, { fullWidth: true, onClick: handleClick, secondary: true }, "Text"));
};
export const SecondaryFullWidthLongText = function () {
    return (h(Button, { fullWidth: true, onClick: handleClick, secondary: true }, longText));
};
export const Destructive = function () {
    return (h(Button, { destructive: true, onClick: handleClick }, "Text"));
};
export const DestructiveLoading = function () {
    return (h(Button, { destructive: true, loading: true, onClick: handleClick }, "Text"));
};
export const DestructiveDisabled = function () {
    return (h(Button, { destructive: true, disabled: true, onClick: handleClick }, "Text"));
};
export const DestructiveDisabledLoading = function () {
    return (h(Button, { destructive: true, disabled: true, loading: true, onClick: handleClick }, "Text"));
};
export const DestructiveFullWidth = function () {
    return (h(Button, { destructive: true, fullWidth: true, onClick: handleClick }, "Text"));
};
export const DestructiveFullWidthLongText = function () {
    return (h(Button, { destructive: true, fullWidth: true, onClick: handleClick }, longText));
};
export const DestructiveSecondary = function () {
    return (h(Button, { destructive: true, onClick: handleClick, secondary: true }, "Text"));
};
export const DestructiveSecondaryLoading = function () {
    return (h(Button, { destructive: true, loading: true, onClick: handleClick, secondary: true }, "Text"));
};
export const DestructiveSecondaryDisabled = function () {
    return (h(Button, { destructive: true, disabled: true, onClick: handleClick, secondary: true }, "Text"));
};
export const DestructiveSecondaryDisabledLoading = function () {
    return (h(Button, { destructive: true, disabled: true, loading: true, onClick: handleClick, secondary: true }, "Text"));
};
export const DestructiveSecondaryFullWidth = function () {
    return (h(Button, { destructive: true, fullWidth: true, onClick: handleClick, secondary: true }, "Text"));
};
export const DestructiveSecondaryFullWidthLongText = function () {
    return (h(Button, { destructive: true, fullWidth: true, onClick: handleClick, secondary: true }, longText));
};
