// create-figma-plugin/lib/settings/create-default-settings.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createDefaultSettings = void 0;
const common_1 = require("@create-figma-plugin/common");
const gitUserName = require("git-user-name");
const create_plugin_display_name_1 = require("./create-plugin-display-name");
function createDefaultSettings(options) {
    const { name, template } = options;
    const author = gitUserName();
    return {
        author: author === null ? undefined : author,
        displayName: typeof name === 'undefined'
            ? common_1.constants.packageJson.defaultPluginDisplayName
            : create_plugin_display_name_1.createPluginDisplayName(name),
        license: common_1.constants.packageJson.defaultLicense,
        name: typeof name === 'undefined'
            ? common_1.constants.packageJson.defaultPluginName
            : name,
        template: typeof template === 'undefined' ? common_1.constants.defaultTemplate : template,
        version: common_1.constants.packageJson.defaultVersion
    };
}
exports.createDefaultSettings = createDefaultSettings;
