// @create-figma-plugin/ui/lib/components/file-upload-button/file-upload-button.stories.js
/** @jsx h */
import { h } from 'preact';
import { FileUploadButton } from './file-upload-button';
export default { title: 'File Upload Button' };
const longText = Array(100).fill('Text').join(' ');
function handleSelectedFiles(files, event) {
    console.log(files); // eslint-disable-line no-console
    console.log(event); // eslint-disable-line no-console
}
export const Button = function () {
    return (h(FileUploadButton, { multiple: true, onSelectedFiles: handleSelectedFiles }, "Text"));
};
export const ButtonLoading = function () {
    return (h(FileUploadButton, { loading: true, multiple: true, onSelectedFiles: handleSelectedFiles }, "Text"));
};
export const ButtonDisabled = function () {
    return (h(FileUploadButton, { disabled: true, multiple: true, onSelectedFiles: handleSelectedFiles }, "Text"));
};
export const ButtonDisabledLoading = function () {
    return (h(FileUploadButton, { disabled: true, loading: true, multiple: true, onSelectedFiles: handleSelectedFiles }, "Text"));
};
export const ButtonFullWidth = function () {
    return (h(FileUploadButton, { fullWidth: true, multiple: true, onSelectedFiles: handleSelectedFiles }, "Text"));
};
export const ButtonFullWidthLongText = function () {
    return (h(FileUploadButton, { fullWidth: true, multiple: true, onSelectedFiles: handleSelectedFiles }, longText));
};
