// typed-scss-modules/dist/lib/typescript/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var class_names_to_type_definition_1 = require("./class-names-to-type-definition");
exports.bannerTypeDefault = class_names_to_type_definition_1.bannerTypeDefault;
exports.classNamesToTypeDefinitions = class_names_to_type_definition_1.classNamesToTypeDefinitions;
exports.exportTypeDefault = class_names_to_type_definition_1.exportTypeDefault;
exports.exportTypeInterfaceDefault = class_names_to_type_definition_1.exportTypeInterfaceDefault;
exports.exportTypeNameDefault = class_names_to_type_definition_1.exportTypeNameDefault;
exports.quoteTypeDefault = class_names_to_type_definition_1.quoteTypeDefault;
exports.EXPORT_TYPES = class_names_to_type_definition_1.EXPORT_TYPES;
exports.QUOTE_TYPES = class_names_to_type_definition_1.QUOTE_TYPES;
var alerts_1 = require("../core/alerts");
exports.logLevelDefault = alerts_1.logLevelDefault;
exports.LOG_LEVELS = alerts_1.LOG_LEVELS;
var get_type_definition_path_1 = require("./get-type-definition-path");
exports.getTypeDefinitionPath = get_type_definition_path_1.getTypeDefinitionPath;
