// @create-figma-plugin/ui/lib/components/layer/layer.stories.js
/** @jsx h */
import { h } from 'preact';
import { useState } from 'preact/hooks';
import { Layer } from './layer';
export default { title: 'Layer' };
const longText = Array(100).fill('Text').join(' ');
export const Frame = function () {
    const [selected, setSelected] = useState(false);
    function handleClick() {
        setSelected(!(selected === true));
    }
    return (h(Layer, { onClick: handleClick, selected: selected, type: "frame" }, "Text"));
};
export const FrameSelected = function () {
    const [selected, setSelected] = useState(true);
    function handleClick() {
        setSelected(!(selected === true));
    }
    return (h(Layer, { onClick: handleClick, selected: selected, type: "frame" }, "Text"));
};
export const FrameWithPageName = function () {
    const [selected, setSelected] = useState(false);
    function handleClick() {
        setSelected(!(selected === true));
    }
    return (h(Layer, { onClick: handleClick, pageName: "Page Name", selected: selected, type: "frame" }, "Text"));
};
export const FrameLongText = function () {
    const [selected, setSelected] = useState(false);
    function handleClick() {
        setSelected(!(selected === true));
    }
    return (h(Layer, { onClick: handleClick, pageName: "Page Name", selected: selected, type: "frame" }, longText));
};
export const Component = function () {
    const [selected, setSelected] = useState(false);
    function handleClick() {
        setSelected(!(selected === true));
    }
    return (h(Layer, { onClick: handleClick, selected: selected, type: "component" }, "Text"));
};
export const ComponentSelected = function () {
    const [selected, setSelected] = useState(true);
    function handleClick() {
        setSelected(!(selected === true));
    }
    return (h(Layer, { onClick: handleClick, selected: selected, type: "component" }, "Text"));
};
export const ComponentWithPageName = function () {
    const [selected, setSelected] = useState(false);
    function handleClick() {
        setSelected(!(selected === true));
    }
    return (h(Layer, { onClick: handleClick, pageName: "Page Name", selected: selected, type: "component" }, "Text"));
};
export const ComponentLongText = function () {
    const [selected, setSelected] = useState(false);
    function handleClick() {
        setSelected(!(selected === true));
    }
    return (h(Layer, { onClick: handleClick, pageName: "Page Name", selected: selected, type: "component" }, longText));
};
