// @create-figma-plugin/ui/lib/components/selectable-item/selectable-item.stories.js
/** @jsx h */
import { h } from 'preact';
import { useState } from 'preact/hooks';
import { SelectableItem } from './selectable-item';
export default { title: 'Selectable Item' };
const longText = Array(100).fill('Text').join(' ');
export const Default = function () {
    const [state, setState] = useState({ foo: false });
    return (h(SelectableItem, { name: "foo", onChange: setState, value: state.foo }, "Text"));
};
export const Selected = function () {
    const [state, setState] = useState({ foo: true });
    return (h(SelectableItem, { name: "foo", onChange: setState, value: state.foo }, "Text"));
};
export const Disabled = function () {
    const [state, setState] = useState({ foo: false });
    return (h(SelectableItem, { disabled: true, name: "foo", onChange: setState, value: state.foo }, "Text"));
};
export const DisabledSelected = function () {
    const [state, setState] = useState({ foo: true });
    return (h(SelectableItem, { disabled: true, name: "foo", onChange: setState, value: state.foo }, "Text"));
};
export const Bold = function () {
    const [state, setState] = useState({ foo: false });
    return (h(SelectableItem, { bold: true, name: "foo", onChange: setState, value: state.foo }, "Text"));
};
export const Indent = function () {
    const [state, setState] = useState({ foo: false });
    return (h(SelectableItem, { indent: true, name: "foo", onChange: setState, value: state.foo }, "Text"));
};
export const LongText = function () {
    const [state, setState] = useState({ foo: true });
    return (h(SelectableItem, { name: "foo", onChange: setState, value: state.foo }, longText));
};
