// @create-figma-plugin/ui/lib/components/columns/columns.stories.js
/** @jsx h */
import { h } from 'preact';
import { Text } from '../text/text';
import { Columns } from './columns';
export default { title: 'Columns' };
const style = {
    backgroundColor: '#daebf7'
};
export const NoSpace = function () {
    return (h(Columns, null,
        h("div", { style: style },
            h(Text, null, "foo")),
        h("div", { style: style },
            h(Text, null, "bar")),
        h("div", { style: style },
            h(Text, null, "baz"))));
};
export const ExtraSmallSpace = function () {
    return (h(Columns, { space: "extraSmall" },
        h("div", { style: style },
            h(Text, null, "foo")),
        h("div", { style: style },
            h(Text, null, "bar")),
        h("div", { style: style },
            h(Text, null, "baz"))));
};
export const SmallSpace = function () {
    return (h(Columns, { space: "small" },
        h("div", { style: style },
            h(Text, null, "foo")),
        h("div", { style: style },
            h(Text, null, "bar")),
        h("div", { style: style },
            h(Text, null, "baz"))));
};
export const MediumSpace = function () {
    return (h(Columns, { space: "medium" },
        h("div", { style: style },
            h(Text, null, "foo")),
        h("div", { style: style },
            h(Text, null, "bar")),
        h("div", { style: style },
            h(Text, null, "baz"))));
};
export const LargeSpace = function () {
    return (h(Columns, { space: "large" },
        h("div", { style: style },
            h(Text, null, "foo")),
        h("div", { style: style },
            h(Text, null, "bar")),
        h("div", { style: style },
            h(Text, null, "baz"))));
};
export const ExtraLargeSpace = function () {
    return (h(Columns, { space: "extraLarge" },
        h("div", { style: style },
            h(Text, null, "foo")),
        h("div", { style: style },
            h(Text, null, "bar")),
        h("div", { style: style },
            h(Text, null, "baz"))));
};
